cl/lib/email_backends.py

```diff
@@ -18,6 +21,93 @@
 )
 
 
+def incr_email_temp_counter(r: Redis) -> None:
+    """increments the temporary counter and adds a new
+    element to the sorted set once it reaches the value of
+    the EMAILS_TEMP_COUNTER setting.
+
+    Args:
+        r (Redis): The Redis DB connection interface.
+    """
+    temp_counter = r.get("email:temp_counter")
+
+    if temp_counter:
+        if int(temp_counter) + 1 >= settings.EMAIL_MAX_TEMP_COUNTER:
+            current_time = time.time_ns()
+            pipe = r.pipeline()
+            pipe.zadd(
+                "email:delivery_attempts", {str(current_time): current_time}
+            )
+            pipe.set("email:temp_counter", 0)
+            pipe.execute()
+            return
+
+    pipe = r.pipeline()
+    pipe.incr("email:temp_counter")
+    pipe.execute()
```

Comment:
This doesn't need a pipeline, does it?

Reply:
No, it doesn't. It's just a single command. Thank you for bringing this to my attention!